extensions/wikia/Flags/maintenance/MoveNotices.php

```diff
@@ -0,0 +1,465 @@
+<?php
+
+$dir = dirname( __FILE__ ) . "/../../../../";
+$extDir = __DIR__ . "/../";
+
+require_once( $dir . 'includes/wikia/nirvana/WikiaObject.class.php' );
+require_once( $dir . 'includes/wikia/nirvana/WikiaModel.class.php' );
+require_once( $extDir . 'FlagsExtractor.class.php' );
+require_once( $extDir . 'models/FlagsBaseModel.class.php' );
+require_once( $extDir . 'models/Flag.class.php' );
+require_once( $extDir . 'models/FlagType.class.php' );
+require_once( $dir . 'maintenance/Maintenance.php' );
+
+use Flags\FlagsExtractor;
+use Flags\Models\Flag;
+use Flags\Models\FlagType;
+
+class MoveNotices extends Maintenance {
+
+	private
+		$log = '',
+		$logFile,
+		$templateName,
+		$wikiId,
+		$pageId,
+		$flagId;
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( 'csv', 'CSV file with data' );
+		$this->addOption( 'list', 'Run script without adding data to database' );
+		$this->addOption( 'section', 'Search template in given section. Otherwise all article is searched.' );
+	}
+
+	public function execute() {
+		global $wgCityId, $wgParser;
+
+		if ( empty ( $wgCityId ) ) {
+			exit( "Wiki ID is not set\n" );
+		}
+
+		$this->wikiId = $wgCityId;
+
+		$csv = $this->getOption( 'csv' );
+		$section = $this->getOption( 'section' );
+		$list = $this->getOption( 'list' );
+
+		if ( empty( $csv ) ) {
+			$this->output( "You must attach CSV file.\n" );
+			return;
+		}
+
+		if ( !file_exists( $csv ) ) {
+			$this->output( "File $csv does not exist.\n" );
+			return;
+		}
+
+		$this->prepareLogFile( $csv );
+
+		if ( !$this->logFile ) {
+			$this->output( "Cannot create log file.\n" );
+			return;
+		}
+
+		$csvFile = fopen( $csv, 'r' );
+		$csvData = [];
+
+		if ( !$csvFile ) {
+			$this->output( "Cannot read file: $csv" );
+			return;
+		}
+
+		// Get data from scv file
+		while( ( $data = fgetcsv( $csvFile ) ) !== false ) {
+			$csvData[] = $data;
+		}
+
+		fclose( $csvFile );
+
+		$this->output( "Start processing\n" );
+
+		$flagModel = new Flag();
+		$flagTypeModel = new FlagType();
+		$flagsExtractor = new FlagsExtractor();
+
+		foreach( $csvData as $data ) {
+			$this->log = '';
+			$flagId = null;
+
+			$error = $this->validateCSVData( $data );
+
+			if ( $error ) {
+				fwrite($this->logFile, $this->log);
+				$this->output( $this->log );
+				continue;
+			}
+
+			$this->templateName = $data[0];
+
+			if ( !$list ) {
+				// Prepare data to add flag type
+				$flagType = $this->prepareDataForFlagType( $flagTypeModel, $data );
+
+				if ( !($this->flagId = $this->addFlagType( $flagTypeModel, $flagType ) ) ) {
+					continue;
+				}
+
+				$this->log = "Adding flag type: " . json_encode( $flagType ) ."\n";
+			}
+
+			$this->log .= "Start processing template: $this->templateName \n";
```

Comment:
A second idea - since we output everything that is logged, maybe do both actions - `$this->log .= ` and `$this->output()` in this method?